forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -178,57 +186,105 @@ pub async fn deploy(command: cmd::Deploy) -> Result<Vec<DeployedContract>> {
         None
     };
 
+    // Ensure that all packages are being deployed to the same node.
+    let node_url = get_node_url(
+        &command.node,
+        &pkgs_to_deploy[0].descriptor.manifest_file.network,
+    )?;
+    if !pkgs_to_deploy.iter().all(|pkg| {
+        get_node_url(&command.node, &pkg.descriptor.manifest_file.network).ok()
+            == Some(node_url.clone())
+    }) {
+        bail!("All contracts in a deployment should be deployed to the same node. Please ensure that the network specified in the Forc.toml files of all contracts is the same.");
+    }
+
+    // Confirmation step. Summarize the transaction(s) for the deployment.
+    let (provider, signing_key) =
+        confirm_transaction_details(&pkgs_to_deploy, &command, node_url.clone()).await?;
+
+    for pkg in pkgs_to_deploy {
+        let salt = match (&contract_salt_map, command.default_salt) {
+            (Some(map), false) => {
+                if let Some(salt) = map.get(pkg.descriptor.manifest_file.project_name()) {
+                    *salt
+                } else {
+                    Default::default()
+                }
+            }
+            (None, true) => Default::default(),
+            (None, false) => rand::random(),
+            (Some(_), true) => {
+                bail!("Both `--salt` and `--default-salt` were specified: must choose one")
+            }
+        };
+        let contract_id = deploy_pkg(
+            &command,
+            pkg,
+            salt,
+            &provider,
+            &signing_key,
+            node_url.clone(),
+        )
+        .await?;
+        contract_ids.push(contract_id);
+    }
+    Ok(contract_ids)
+}
+
+/// Prompt the user to confirm the transactions required for deployment, as well as the signing key.
+async fn confirm_transaction_details(
```

Comment:
Do we want to support automatic deployments via the CLI that don't require confirmations/interactive use? I realize you could do it yourself via the TS/RS SDKs, but having the CLI handle the details and passing in a bunch of stuff via the command line could be a nice flow for customers who want to use CI/CD (especially for testnets).

Reply:
Yes, this is already supported in the CLI. This PR is just about improving the existing interactive UX, specifically making it so that users don't have to re-enter the same inputs multiple times for multi-tx deployments (laying the groundwork for proxy and/or chunked deployments).